### PR TITLE
Add support for Zuul jobs in operator_build role

### PR DESCRIPTION
**Pull request by viroel**
This patch adds the processing of zuul information to retrieve PR information and add to operator_build list. Update molecule to simulate data from zuul.
Combine operators and meta-operator within the same list

This PR has:
- [x] Appropriate testing (molecule, python unit tests)
- [x] Appropriate documentation (README in the role, main README is up-to-date)

### CODE REVIEW

**Comment by viroel** [comment]
/assign @cjeanner 

@raukadah hi, can you help me to get `zuul` vars available in playbook execution?

File: ci_framework/roles/operator_build/tasks/zuul_info.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        zuul_info_operators: "{{ zuul_info_operators|default([]) + operator_info }}"
+      vars:
+        operator_info:
+          - name: "{{ item.project.short_name }}"
+            src: "{{ ansible_user_dir }}/{{ item.project.src_dir }}"
+            pr_sha: "{{ item.patchset }}"
+            pr_owner: "{{ pr_owner.stdout }}"
+  when:
+    - "'change_url' in item"
+    - "'operator' in item.project.short_name"
```
Comment (raukadah) [inline ci_framework/roles/operator_build/tasks/zuul_info.yml:39]: In end-to-end job, we are calling running [ansible-playbook](https://github.com/openstack-k8s-operators/ci-framework/blob/main/.zuul.d/e2e-run.yml#L6) and the zuul related vars are not getting passed to nested ansible playbook that's why we are not above to discover it.

Comment (raukadah) [reply ci_framework/roles/operator_build/tasks/zuul_info.yml:39]: @viroel https://github.com/openstack-k8s-operators/ci-framework/pull/136 will make zuul vars available to the nested ansible playbooks calls.

Comment (cjeanner) [reply ci_framework/roles/operator_build/tasks/zuul_info.yml:39]: if there's a better way to run the end-to-end, I'd be happy to learn :)

File: ci_framework/roles/operator_build/tasks/zuul_info.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Populate operators from zuul info
+  block:
+    - name: Get the PR owner using github api
+      register: pr_owner
+      ansible.builtin.shell: |
```
Comment (cjeanner) [inline ci_framework/roles/operator_build/tasks/zuul_info.yml:21]: would it be possible to use native ansible? afaik, ansible.builtin.uri may pass the required headers - then, we can extract the json content we need.

File: ci_framework/roles/operator_build/tasks/zuul_info.yml (hunk 1 above)
Comment (viroel) [reply ci_framework/roles/operator_build/tasks/zuul_info.yml:40]: thanks for fixing this

File: ci_framework/roles/operator_build/tasks/zuul_info.yml (hunk 2 above)
Comment (viroel) [reply ci_framework/roles/operator_build/tasks/zuul_info.yml]: yeah, much better, and we don't need to use 'jq' Thanks

File: ci_framework/roles/operator_build/tasks/main.yml [hunk 3]
```diff
@@ -14,52 +14,66 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-- name: Ensure output directory exists
+- name: Ensure mandatory directories exist
   ansible.builtin.file:
-    path: "{{ cifmw_operator_build_basedir }}/artifacts"
+    path: "{{ cifmw_operator_build_basedir }}/{{ item }}"
     state: directory
+  with_items:
```
Comment (cjeanner) [inline ci_framework/roles/operator_build/tasks/main.yml:21]:
```suggestion
  loop:
```

File: ci_framework/roles/operator_build/molecule/default/converge.yml [hunk 4]
```diff
@@ -21,27 +21,49 @@
...
+          - cifmw_operator_build_output['operators']['keystone-operator']['image_storage_bundle'] is defined
+          - cifmw_operator_build_output['operators']['keystone-operator']['image_catalog'] is defined
           # Check the meta operator build output
           - cifmw_operator_build_output['operators']['openstack-operator']['commit_hash'] is defined
           - cifmw_operator_build_output['operators']['openstack-operator']['image'] is defined
           - cifmw_operator_build_output['operators']['openstack-operator']['image_bundle'] is defined
           - cifmw_operator_build_output['operators']['openstack-operator']['image_storage_bundle'] is defined
           - cifmw_operator_build_output['operators']['openstack-operator']['image_catalog'] is defined
+
+    - name: Build mariadb-operator from Zuul job
```
Comment (cjeanner) [inline ci_framework/roles/operator_build/molecule/default/converge.yml:49]: Care to assert we get the related content in the cifmw_operator_build_output? Just to be on the safest possible side

File: ci_framework/roles/operator_build/tasks/main.yml [hunk 5]
```diff
@@ -14,52 +14,66 @@
...
   vars:
     operator: "{{ item }}"
   ansible.builtin.include_tasks: build.yml
-  loop: "{{ cifmw_operator_build_operators }}"
+  loop: "{{ operators_list }}"
 
 - name: Building meta operator
   when:
     - cifmw_operator_build_meta_build is true
+    - "item.name == cifmw_operator_build_meta_name"
```
Comment (cjeanner) [inline ci_framework/roles/operator_build/tasks/main.yml:72]: afaik no need to use quotes here

**Review by cjeanner — CHANGES_REQUESTED** [review]

**Review by marios — APPROVED** [review]
stared for a bit

seems reasonable

**Pull request merged**